src/hotspot/share/gc/shared/barrierSetNMethod.cpp

```diff
@@ -181,7 +181,7 @@ void BarrierSetNMethod::arm_all_nmethods() {
 int BarrierSetNMethod::nmethod_stub_entry_barrier(address* return_address_ptr) {
   // Enable WXWrite: the function is called directly from nmethod_entry_barrier
   // stub.
-  MACOS_AARCH64_ONLY(ThreadWXEnable wx(WXWrite, Thread::current()));
+  MACOS_AARCH64_ONLY(ThreadWXEnable wx(DefaultWXWriteMode, Thread::current()));
```

Comment:
In 8361376 I have moved this down into BarrierSetNMethod::nmethod_entry_barrier.